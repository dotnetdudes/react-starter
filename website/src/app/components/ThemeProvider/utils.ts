import { ThemeType } from './slice/types';

/* istanbul ignore next line */
export const isSystemDark = window?.matchMedia
  ? window.matchMedia('(prefers-color-scheme: dark)')?.matches
  : undefined;

export function saveTheme(theme: ThemeType) {
  window.localStorage && localStorage.setItem('selectedTheme', theme);
}

/* istanbul ignore next line */
export function getThemeFromStorage(): ThemeType | null {
  return window.localStorage
    ? (localStorage.getItem('selectedTheme') as ThemeType) || null
    : null;
}
