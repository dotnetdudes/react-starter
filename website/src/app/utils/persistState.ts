// @flow
// https://egghead.io/lessons/javascript-redux-persisting-the-state-to-the-local-storage
// https://medium.com/@jrcreencia/persisting-redux-state-to-local-storage-f81eb0b90e7e
import type { RootState } from 'types';

export const loadState = () => {
  try {
    const serializedState = sessionStorage.getItem('stateSramReg');
    if (serializedState === null) {
      return undefined;
    }
    // $FlowIgnore
    return JSON.parse(serializedState);
  } catch (err) {
    return undefined;
  }
};

export const saveState = (state: RootState) => {
  try {
    const serializedState = JSON.stringify(state);
    sessionStorage.setItem('stateSramReg', serializedState);
  } catch (err) {
    // die
  }
};
