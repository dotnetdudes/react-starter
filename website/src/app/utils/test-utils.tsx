import React, { FC, ReactElement } from 'react';
import { render, RenderOptions } from '@testing-library/react';
import { Provider } from 'react-redux';
import { BrowserRouter } from 'react-router-dom';
import { configureAppStore } from 'store/configureStore';
import { ThemeProvider, createTheme } from '@mui/material/styles';

const store = configureAppStore();

const MockTheme = ({ children }: any) => {
  const theme = createTheme({});
  return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
};

const AllTheProviders: FC<{ children: React.ReactNode }> = ({ children }) => {
  return (
    <Provider store={store}>
      <MockTheme>
        <BrowserRouter>{children}</BrowserRouter>
      </MockTheme>
    </Provider>
  );
};

const customRender = (
  ui: ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
) => render(ui, { wrapper: AllTheProviders, ...options });

export * from '@testing-library/react';
export { customRender as render };
