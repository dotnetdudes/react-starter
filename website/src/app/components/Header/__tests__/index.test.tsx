import { Header } from '../index';
import { render } from 'app/utils/test-utils';

jest.mock('react-i18next', () => ({
  useTranslation: () => {
    return {
      t: str => str,
      i18n: {
        changeLanguage: () => new Promise(() => {}),
      },
    };
  },
}));

describe('<Header  />', () => {
  it('should match snapshot', () => {
    const header = render(<Header />);
    expect(header.container.firstChild).toMatchSnapshot();
  });

  it('should render a <div> tag', () => {
    const header = render(<Header />);
    expect(header.container.querySelector('div')).toBeInTheDocument();
  });
});