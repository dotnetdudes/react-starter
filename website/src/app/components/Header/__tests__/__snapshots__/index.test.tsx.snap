// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Header  /> should match snapshot 1`] = `
<header
  class="MuiPaper-root MuiPaper-elevation MuiPaper-elevation0 MuiAppBar-root MuiAppBar-colorPrimary MuiAppBar-positionSticky css-1uwl3xt-MuiPaper-root-MuiAppBar-root"
>
  <div
    class="MuiToolbar-root MuiToolbar-gutters MuiToolbar-dense css-10ltm04-MuiToolbar-root"
  >
    Header.heading
    <button
      aria-label="Theme.selectDark"
      class="MuiButtonBase-root MuiIconButton-root MuiIconButton-colorInherit MuiIconButton-sizeMedium css-zylse7-MuiButtonBase-root-MuiIconButton-root"
      tabindex="0"
      type="button"
      value="dark"
    >
      <svg
        aria-hidden="true"
        class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
        data-testid="Brightness4Icon"
        focusable="false"
        viewBox="0 0 24 24"
      >
        <path
          d="M20 8.69V4h-4.69L12 .69 8.69 4H4v4.69L.69 12 4 15.31V20h4.69L12 23.31 15.31 20H20v-4.69L23.31 12 20 8.69zM12 18c-.89 0-1.74-.2-2.5-.55C11.56 16.5 13 14.42 13 12s-1.44-4.5-3.5-5.45C10.26 6.2 11.11 6 12 6c3.31 0 6 2.69 6 6s-2.69 6-6 6z"
        />
      </svg>
      <span
        class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
      />
    </button>
  </div>
</header>
`;
